### ISSUE TITLE
Unknown Inverter Type message

### ISSUE THREAD

**Issue by SBFspot** [SBFspot/SBFspot#118]
Hi,

i am getting a Unknown Inverter Type message. Is this the reason there is no output to the csv file?

Details:

0x00002455 and Inverter Type=￼Sunny Boy 1.5

detail product information: http://www.sma-benelux.com/nl_BE/producten/omvormers/sunny-boy/sunny-boy-15-25.html

#### Attachments
[TagListEN-US.zip](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=113&FileAttachmentId=1461460)
[logging.txt](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=113&FileAttachmentId=1461627)
[logging with -v5 -d5.txt](https://www.codeplex.com/Download/AttachmentDownload.ashx?ProjectName=SBFspot&WorkItemId=113&FileAttachmentId=1461628)

#### This work item was migrated from CodePlex
CodePlex work item ID: '113'
Vote count: '2'

**Comment by SBFspot** [comment]
[mtijssen@9/06/2015]
sudo /usr/local/bin/sbfspot.3/SBFspot -sp0 -v
SBFspot V3.0.8
Yet another tool to read power production of SMA solar inverters
(c) 2012-2015, SBF (https://sbfspot.codeplex.com)
Compiled for Linux (LE) 32 bit

Commandline Args: -sp0 -v
Reading config '/usr/local/bin/sbfspot.3/SBFspot.cfg'
Tue Jun  9 14:01:24 2015: INFO: Starting...
sunrise: 05:20
sunset : 21:49
Connecting to Local Network...
Initializing...
SUSyID: 125 - SessionID: 981977752 (0x3A87CA98)
Inverter IP address: XXXXXXXXXX from SBFspot.cfg
Logon OK
SetInverterTime() not executed! We're still testing...
Unknown Inverter Type. Report this issue at https://sbfspot.codeplex.com/workitem/list/basic with following info:
0x00002455 and Inverter Type=<Fill in the exact type> (e.g. SB1300TL-10)
SUSyID: 303 - SN: XXXXXXXXX
Device Name:      SB1.5
Device Class:     Solar Inverters
Device Type:      UNKNOWN TYPE
Software Version: 02.03.01.R
Serial number:    XXXXXXXXX
SUSyID: 303 - SN: XXXXXXXXX
Device Status:      Ok
SUSyID: 303 - SN: XXXXXXXXX
Device Temperature: 58.0°C
SUSyID: 303 - SN: XXXXXXXXX
GridRelay Status:      Closed
SUSyID: 303 - SN: XXXXXXXXX
Pac max phase 1: 1500W
Pac max phase 2: 0W
Pac max phase 3: 0W
SUSyID: 303 - SN: XXXXXXXXX
Energy Production:
        EToday: 4.013kWh
        ETotal: 4.016kWh
        Operation Time: 8.88h
        Feed-In Time  : 8.63h
SUSyID: 303 - SN: XXXXXXXXX
DC Spot Data:
        String 1 Pdc:   0.560kW - Udc: 203.32V - Idc:  2.757A
        String 2 Pdc:   0.000kW - Udc:   0.00V - Idc:  0.000A
SUSyID: 303 - SN: XXXXXXXXX
AC Spot Data:
        Phase 1 Pac :   0.539kW - Uac: 229.21V - Iac:  2.355A
        Phase 2 Pac :   0.000kW - Uac:   0.00V - Iac:  0.000A
        Phase 3 Pac :   0.000kW - Uac:   0.00V - Iac:  0.000A
        Total Pac   :   0.539kW
SUSyID: 303 - SN: XXXXXXXXX
Grid Freq. : 49.97Hz
SUSyID: 303 - SN: XXXXXXXXX
Current Inverter Time: 09/06/2015 14:01:23
Inverter Wake-Up Time: 09/06/2015 05:36:07
Inverter Sleep Time  : 09/06/2015 14:01:22
********************
* ArchiveDayData() *
********************
startTime = 55761060 -> 09/06/2015 00:00:00
ArchiveDayData returned an error: -1
**********************
* ArchiveMonthData() *
**********************
startTime = 556C3B30 -> 01/06/2015 13:00:00
ExportMonthDataToCSV()
There is no data to export!
Reading events: 2015-Jun-01
ArchiveEventData(user) returned an error: -1
Tue Jun  9 14:01:40 2015: INFO: Done.

**Comment by SBFspot** [comment]
[SBF@9/06/2015]
Error -1 means there is no data. Not sure, but it's possible this inverter doesn't keep the historic data.
Did you try with Sunny Explorer? What version? (I think 1.7.20 is the latest)
Are you able to visualize and export (csv) day/month data with SE?

**Comment by SBFspot** [comment]
[mtijssen@10/06/2015]
Hi SBF,

Sunny Explorer has also a problem with this inverter when only exporting day and month there are no files, but when exporting day/month and events there are files.

I already have send an e-mail to SMA to have a look and hopefully solve this problem by firmware update.

The unknown inverter message is still an active issue.

**Comment by SBFspot** [comment]
[SBF@10/06/2015]
So it is possible to get the historic data. One thing to check first: is the time of your inverter correct? If this is OK, retry with SBFspot with debug enabled (-v5 -d5) This will give us more info.
Firmware will be OK I think as it is a recent model, but SMA should release a new version of SE. As far as I can see, SB 1.5 is not recognized by SE 1.07.20R
Replace the file TaglistEN-US.txt with the attached one and SBFspot will show you the correct type.

**Comment by SBFspot** [comment]
[mtijssen@11/06/2015]
Attached is the output of the debug mode.

SMA is already looking into the problem of the data export as I send them some screenshots with the explanation. Waiting for their response and will give an update here.

Replacement of the TaglistEN-US.txt file solved the issue with unknown inverter type.

**Comment by SBFspot** [comment]
[mtijssen@11/06/2015]
loggin with -v5 -d5

**Comment by SBFspot** [comment]
[SBF@21/06/2015]
Any update from SMA?

**Comment by SBFspot** [comment]
[mtijssen@22/06/2015]
Not yet. 

Last reaction from SMA last week. 

The person that needs to look into this problem is still on vacation.

But exports are also go wrong with the newest Sunny Explorer when you don't export the data without the events. 

Will update this issue when I have received a new reaction from SMA.

**Comment by SBFspot** [comment]
[SBF@3/07/2015]
There is a new Sunny Explorer (1.07.23) released: http://www.sma-benelux.com/nl_BE/producten/software/sunny-explorer.html

Can you check the export with this version? Also, I'm curious if it recognizes your Sunny Boy 1.5

**Comment by SBFspot** [comment]
[mtijssen@3/07/2015]
Unfortunately it's still not working. Export is till going wrong and it doesn't recognize the SB 1.5

I also have send SMA a reminder yesterday to remind them about this issue, but haven't had a reaction. So still waiting.

**Comment by SBFspot** [comment]
[Pascal69480@28/07/2015]
Hi,
I'm really interrested by this post because I've the same problem with a Sunny Boy 1.5 : errors while exporting csv files with SE (release 1.07.23), unless I check the 'event' checkbox

I also have tried to share data with Sunny Portal but I can't get the ID representing my plant.
The support on SMA website [told me to call the french Service Line](http://en.sma-sunny.com/2015/01/06/service-tip-what-if-the-connection-to-sunny-portal-doesnt-work/) 

See also [here](https://sbfspot.codeplex.com/discussions/642077)

**Comment by SBFspot** [comment]
[SillieWimons@29/07/2015]
dear all;   the below info is about the insertion of the "new" value for this inverter in the tag files
!!! this should only help to eleminate the message "unknown" inverter type:
!!! this does not solve the problem "-1" no data

-a- about the adaptations for the tag file 
the tag file "en-us" adapated by s.b.f. for this issue C113 contains a new line 
these tags are decimal numbered - 0x0002455 -> 0d009301 

when inserting the below line (cfr a.1) in other languages tag files
then your type of inverter should be recognized (when using another language as english)
 
the correct place is after the line with tag = 9295   AND    before the line with tag = 9500
 
be aware: use a "text" editor and not "WORD" or other -> the resulting file must be a plain txt file
 -a.1- between the quotes is the text for the new line -> "9301=SB 1.5\0\SB 1.5"

if in doubt look into the english version file 

kind regards   wim

**Comment by SBFspot** [comment]
[Pascal69480@29/07/2015]
Thank you for this explanation !
All is fine when adding this line in my french tag files (using Notepad++).
```
Wed Jul 29 23:51:08 2015: INFO: Starting...
sunrise: 04:56
sunset : 17:09
Connecting to Local Network...
Initializing...
SUSyID: 125 - SN: xxxxxxxx (xxxxxxxx)
Inverter IP address: 192.168.0.5 from SBFspot.cfg
Logon OK
SetInverterTime() not executed! We're still testing...
SUSyID: 303 - SN: 1930002676
Device Name:      SB1.5-1VL-40 676
Device Class:     Onduleur solaire
Device Type:      SB 1.5
Software Version: 02.03.01.R
```

By the way, the first time I tried ton launch SBFSpot, Sunny Explorer was open and I cannot connect to the inverter.
After closing the program, all was OK.

**Comment by SBFspot** [comment]
[SillieWimons@30/07/2015]
dear mthijssen and pascal,  

in the documentation - SMA states that your inverter type includes "new" communication options
the knwoledge base of SBFspot is the result of reverse engineering,

one of the needed steps for understanding the "new" options is that "SBFspot" users cooperate
 
can you deliver the following information, this will help to obtain better understanding:
?a-1? which is the communication setup that is used between your inverter and your system
(system: RPi or PC or . . . )
?a-2? which are the computer-systems that you use,
you mention the usage of SunExplorer (which runs on windows)

rational: SMA indicates that te new inverter can work directly with WiFi
there are some difference between "data" protocols (depending on the chain used)

?b? are there other units that are part of your system (aside of inverter and computer-system
examples: homemanager, . . . 
rational: as indicated by pascal, he can only use either SunExplorer or SBFspot
!! there are two limits on the comms between inverter and SBFspot
?b-1? on one computer system you can only run OR SunExplorer OR SBFspot
because the drrver software can only be in use by one program
?b-2? anotherc known limit in SMAsystems is that inverters allow only two "sessions" in parallel
so if "home-manager is active -> only one extra system can access the inverter
 
?c? till now i could only find "one" log file (the one of mthijssen)
may i ask to execute the following variants of commands and deliver the log-files individual (or zipped)
for syntax 'use -? help info from sbfspot on the options
for pascal -> please replace "mthijs" by "pascal"
?c.1? SBFspot -sp0 -ad0 -am1 -ae0 -d5 -v5 1>mthijs-log-onlymonth.txt
?c.2? SBFspot -sp0 -ad0 -am0 -ae1 -d5 -v5 1>mthijs-log-onlyevent.txt
?c.3? SBFspot -sp0 -ad0 -am0 -ae1 -installer -password1111 -d5 -v5 1>mthijs-log-onlymonth.txt

a special one - since the first time out occurs in the part "archive" daily: 
can you exectue this command EARLY MORNING - I mean before 6h00 (but after 01h30)
?c53? SBFspot -sp0 -ad1 -am0 -ae0 -d5 -v5 1>mthijs-log-earlyday.txt
rational: if SMA does use an upgraded comms version, one might get some more info

kind regards wim

**Comment by SBFspot** [comment]
[Pascal69480@30/07/2015]
Hello,
As then Sunny Boy 1.5 already provides an integrated webconnect system, there's no additional peripheral needed. The inverter directly sends the data to SunnyPortal.
The inverter communicates with SMA DB through a Wifi connection (inverter -> internet box -> SMA DB)
But I don't like the charts they give and would like to collect the data on my personal computer
That's why I'd like to recover the production wether in SMA DB (with SunnyPortal) or in inverter internal storage (with Sunny Explorer).

?a-1? PC 
?a-2?  Windows Seven OS, connection : PC (eth) -> box -> (Wifi) inverter
As said before, I generate CSV files with SunnyPortal or with SunnyExplorer

?b? Not at all
?b-1? That is was I've noticed yesterday :)
?b-2? Maybe when you try to reach the inverter from SunnyPortal (or SunnyPlaces), or when the inverter is sending data to SMA DB (I don't know the frequency), it already uses 1 session from the 2 parallels allowed
I'll try tonight to see if there is a the difference when I've got an input connection from internet to the inverter or not

?c? I will do it when I'm at home :)

Thank you for your help, and I hope I'll be able to give you useful informations back !

**Comment by SBFspot** [comment]
[mtijssen@30/07/2015]
Hi Pascal/Wim,

after a couple a e-mails and a call to the Dutch SMA Service Line in Belgium, I hope they understand the situation/problem. 

First they mention is that the SB 1.5 has it's own web interface to see the stats or you can see them thought Sunnyportal. After that they mentioned that the SB 1.5 would not support Sunny Explorer, but this is not what is mentioned on their website.

It looks like they didn't test exporting only day or month values. Because Sunny Portal and Sunny Explorer, also catch the event logging they don't receive the error. This is also what can be reproduced with Sunny Explorer. Exporting with the event log disabled will give you an error message. Exporting with the event log is going without any problems and all files are generated. (so a solution would be also ask for the eventlog, but do nothing with it of they don't fix the problem with a firmware update)

The Service Line couldn't tell me if this export methode is correct or if there is a failure with this ' feature' . 

Sunny Explorer itself also has a command line options, but only allow you to get the day / month / event with separate commands, so that also can't be used. 

My setup is that I use SBFspot on a RPi, but because of the problems and SMA uses their own software I have been testing with Windows 2012 Server machine with Sunny Explorer installed.

I will give you the logging when I am at home also.

Regards,
Michael

**Comment by SBFspot** [comment]
[SillieWimons@30/07/2015]
hej  michael and pascal    from info received    i did get extra indications

can you please execute the following additional commands

?c.6? SBFspot -sp0 -ad0 -am3 -ae0 -d5 -v5 1>mthijs-log-only3month.txt
?c.7? SBFspot -sp0 -ad0 -am0 -ae3 -d5 -v5 1>mthijs-log-only3events.txt

thank your for the cooperation   wim

**Comment by SBFspot** [comment]
[mtijssen@1/08/2015]
I have send the requested files by mail.

Regards,
Michael

**Comment by SBFspot** [comment]
[Pascal69480@2/08/2015]
I did the same :)

**Comment by SBFspot** [comment]
**Issue closed by _SBF_ with comment**
Fixed in V3.1.1

**Reason closed**
Fixed
